temporalio/lib/temporalio/worker.rb

```diff
@@ -363,7 +358,6 @@ def initialize(
       activity_executors: ActivityExecutor.defaults,
       workflow_executor: WorkflowExecutor::ThreadPool.default,
       interceptors: [],
-      build_id: Worker.default_build_id,
```

Comment:
Hrmm, now I am wondering how we get the behavior of default build ID back. People expect a default build ID from our SDKs. This makes me wonder if we should revisit deprecating build-ID-only across SDKs since that's what is default for all SDKs today.

I am bringing up the deprecate-build-id-only thing internally again with this concern.